### PR TITLE
Move auth verification to vercel, store state in user context instead of cookies

### PR DESCRIPTION
**Pull request by janiseliang**
will hopefully eliminate dependence on google cloud functions for user authorization, nd also get rid of 404/500 server errors on prod
- moves user storage from cookies into user context (slightly safer since we no longer depend on cookies)
- add "loading..." message when user hasn't loaded on page load (instead of "not authorized")
- removed the App.getInitialProps function in `pages/_app.js` that seems to be causing the 404 errors. apparently this may break CORS requests, but I don't think we're using any... TODO test every API to make sure

TODO:
- clean up checkout, checkin pages that had the override for authToken timeouts?
- test every API to make sure CORS requests (if there are any?) aren't causing issues

### CODE REVIEW

File: pages/checkout.js [hunk 1]
```diff
@@ -235,7 +235,7 @@ class Cart extends React.Component {
   submitCart = async (e) => {
     e.preventDefault();
     this.setState({loading: true})
-    let token = await this.state.user.googleUser.getIdToken()
+    let token = await this.user.googleUser.getIdToken()
```
Comment (justinazhang) [inline pages/checkout.js:238]: Can we change to this.user.authToken

File: pages/checkout.js [hunk 2]
```diff
@@ -483,7 +483,7 @@ class Cart extends React.Component {
             {/* save edit */}
             {this.state.isEditing && <button className='ml-5 text-blue-700 hover:text-blue-500'
                 onClick={async () => {
-                let token = await this.state.user.googleUser.getIdToken()
+                let token = await this.user.googleUser.getIdToken()
```
Comment (justinazhang) [inline pages/checkout.js:486]: This too

**Review by justinazhang — APPROVED** [review]
Yay left some comments but tysm for doing this!! In terms of rollout, curious your thoughts on this: I'll probably test it on my shifts for ~1 week before we merge it into main? and then we'll merge to main if no bugs/anything wrong with some backup prod envs. And then after a month probably just removing all the cloud functions entirely from firebase

**Comment by janiseliang** [comment]
> Yay left some comments but tysm for doing this!! In terms of rollout, curious your thoughts on this: I'll probably test it on my shifts for ~1 week before we merge it into main? and then we'll merge to main if no bugs/anything wrong with some backup prod envs. And then after a month probably just removing all the cloud functions entirely from firebase

that sounds good! yeah a little bit hesitant to release to prod yet so this seems like a good test

**Pull request merged**